docs/admin/etcd.md

```diff
@@ -1,49 +1,196 @@
+   holds that information and is stored in the etcd-data-specific directory
+   (next to the etcd data). If the file doesn’t exist, we default it to 2.2.1
+   version.
+1. If we are in 2.2.1 version and are supposed to upgrade, backup
+   data.
+1. Based on the detected previous etcd version and the desired one (the desired
+   version is communicated via environment variable), do the upgrade steps (if
+   needed).
+   That means, for every minor etcd release between the detected one (excluding)
+   to the desired one (including):
+   1. start etcd in that version
+   1. wait until it is healthy (healthy means that you can write some data to
+      it)
+   1. stop this etcd
+   The important thing here is that those etcd will not listen on the default
+   etcd port.  They are hardcoded to listen on ports which the apiserver is not
+   configured to connect to, which means that apiserver won’t be able to connect
+   to those etcd's.  Assuming no other client goes out of its way to try to
+   connect and write to this obscure port, no new data will be written during
+   this period.
+1. If the desired api to use is v3 and detected one is v2, do the offline
+   migration from v2 to v3 data format.
+   For that we are using two tools:
+   1. ./etcdctl migrate - official tool for migration provided by the etcd team
+   1. custom script that is attaching TTLs to events in the etcd (etcdctl
+      migrate doesn’t support TTLs)
+1. After every successful step, update contents of “version file”.
+   This will protect us from the situation that e.g. something will crash in the
+   meantime and the “version file” will be completely unsynchronized with the
+   real data. Note that it is safe if the script will crash after the step is
+   done and before the file is updated - this will only result in redoing one
+   step in the next try.
+
+All the previous steps are for the case where detected version is earlier or
+equal to the desired one. In the opposite case (i.e. requested rollback), the
+script:
+
+1. Checks if detected version is 3.0.x with v3 API, and desired is 2.2.1 with v2
+   API
+   We don’t support any other rollback.
+1. If so, we run the custom tool provided by etcd team to do the offline
+   rollback.  This tool reads the v3 formatted data and writes it back to disk
+   in v2 format.
+1. Finally update contents of “version file”.
+
+### Upgrade procedure
+Simply modify the command-line in etcd manifest to:
+
+1. Run the migration script
+   If the previously run version is already in the desired version, this will be
+   no-op.
+1. Start etcd (in the desired version)
+
+Starting in 1.6, this has been done in the manifests for new GCE clusters.  You
+should also specify these environment variables.  In particular, **you must keep
+`STORAGE_MEDIA_TYPE` set to application/json if you wish to preserve the option
+to roll back**.
+
+```
+TARGET_STORAGE=etcd3
+ETCD_IMAGE=3.0.17
+TARGET_VERSION=3.0.17
+STORAGE_MEDIA_TYPE=application/json
```

Comment:
are these the GCE defaults or an example of what you would do to preserve the option to roll back?